x/vm/internal/keeper/keeper.go

```diff
@@ -88,39 +88,48 @@ func (k Keeper) ExecuteScriptNoProcessing(ctx sdk.Context, msg types.MsgExecuteS
 func (k Keeper) DeployContract(ctx sdk.Context, msg types.MsgDeployModule) error {
 	k.modulePerms.AutoCheck(types.PermVmExec)
 
-	req, sdkErr := NewDeployRequest(ctx, msg)
-	if sdkErr != nil {
-		return sdkErr
+	execList := make([]*vm_grpc.VMExecuteResponse, len(msg.Module))
+	var intErr error
+
+	for i, contact := range msg.Module {
+		req, sdkErr := NewDeployRequest(ctx, msg.Signer, contact)
+		if sdkErr != nil {
+			return sdkErr
```

Comment:
I think error should has a conext (contract index), otherwise it is hard to debug